scripts/zkai_d128_native_block_gap_accounting_gate.py

```
@@ -0,0 +1,847 @@
+                try:
+                    os.unlink(temp_name, dir_fd=parent_fd)
+                except FileNotFoundError:
+                    pass
+            os.close(parent_fd)
+            raise
+
+    def replace_temp(temp: tuple[pathlib.Path, str, int, tuple[int, int]], path: pathlib.Path, label: str) -> None:
+        _tmp_path, temp_name, parent_fd, identity = temp
+        _assert_directory_identity(path.parent, identity, label)
+        if path.is_symlink():
+            raise D128NativeBlockGapAccountingError(f"{label} must not include symlink components")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)